rpc/types/src/misc/misc.rs

```diff
@@ -0,0 +1,539 @@
+//! Miscellaneous types.
+//!
+//! These are `struct`s that appear in request/response types.
+//! For example, [`crate::json::GetConnectionsResponse`] contains
+//! the [`crate::misc::ConnectionInfo`] struct defined here.
```

Comment:
TODO: I have to document this but `misc.rs` contains all misc types that can be created in a macro-friendly way.

The other files in the `misc/` folder are because they have something special, so the macro doesn't work.